src/main/kotlin/uk/gov/dwp/dataworks/services/TaskDeploymentService.kt

```diff
@@ -48,137 +30,72 @@ class TaskDeploymentService {
-
-        val albRequest = DescribeLoadBalancersRequest.builder().names(albName).build()
-        val albResponse = albClient.describeLoadBalancers(albRequest)
-
-        val albListenerRequest = DescribeListenersRequest.builder().loadBalancerArn(albResponse.loadBalancers()[0].loadBalancerArn()).build()
-        val albListenerResponse = albClient.describeListeners(albListenerRequest)
-
-        val tgRequest = CreateTargetGroupRequest.builder().name("$userName-target-group").protocol("HTTPS").vpcId(albResponse.loadBalancers()[0].vpcId()).port(containerPort).build()
-        val targetGroupResponse = albClient.createTargetGroup(tgRequest)
-
-        logger.info("Created target groups", "target_groups" to targetGroupResponse.targetGroups().joinToString { it.targetGroupName() })
-
-        val albTargetGroupRequest = albClient.describeTargetGroups(DescribeTargetGroupsRequest.builder().names("$userName-target-group").build())
-        val albTargetGroupArn = albTargetGroupRequest.targetGroups()[0].targetGroupArn()
-
-        val pathPattern = PathPatternConditionConfig.builder().values("/$userName/*").build()
-        val albRuleCondition = RuleCondition.builder().field("path-pattern").pathPatternConfig(pathPattern).build()
-        val userTargetGroup = TargetGroupTuple.builder().targetGroupArn(albTargetGroupArn).build()
-        val forwardAction = ForwardActionConfig.builder().targetGroups(userTargetGroup).build()
-        val albRuleAction = Action.builder().type("forward").forwardConfig(forwardAction).build()
+    fun taskDefinitionWithOverride(userName: String, jupyterCpu: Int, jupyterMemory: Int, additionalPermissions: List<String>) {
+        // Retrieve required params from environment
+        val containerPort = Integer.parseInt(configurationResolver.getStringConfig(ConfigKey.USER_CONTAINER_PORT))
+        val emrClusterHostName = configurationResolver.getStringConfig(ConfigKey.EMR_CLUSTER_HOST_NAME)
+        val albName = configurationResolver.getStringConfig(ConfigKey.LOAD_BALANCER_NAME)
+        val ecsClusterName = configurationResolver.getStringConfig(ConfigKey.ECS_CLUSTER_NAME)
+
+        // Load balancer & Routing
+        val loadBalancer = awsCommunicator.getLoadBalancerByName(albName)
+        val listener = awsCommunicator.getAlbListenerByPort(loadBalancer.loadBalancerArn(), containerPort)
+        val targetGroup = awsCommunicator.createTargetGroup(loadBalancer.vpcId(), "$userName-target-group", containerPort)
+        // There are 2 distinct LoadBalancer classes in the AWS SDK - ELBV2 and ECS. They represent the same LB but in different ways.
+        // The following is the load balancer needed to create an ECS service.
+        val ecsLoadBalancer = LoadBalancer.builder()
+                .targetGroupArn(targetGroup.targetGroupArn())
+                .loadBalancerName(loadBalancer.loadBalancerName())
+                .containerName("") //TODO determine which container this will route to
```

Comment:
The container would be named 'guacamole', defined later on line 94.

Reply:
Thanks for clarifying - I wasn't sure whether this would be `guacamole` or `guacd`. I will update